applications/GeoMechanicsApplication/custom_strategies/builder_and_solvers/residualbased_block_builder_and_solver_with_mass_and_damping.h

```diff
@@ -0,0 +1,658 @@
+    KRATOS_DEFINE_LOCAL_FLAG( SILENT_WARNINGS );
+
+    /// Definition of the pointer
+    KRATOS_CLASS_POINTER_DEFINITION(ResidualBasedBlockBuilderAndSolverWithMassAndDamping);
+
+    /// Definition of the base class
+    using BaseType = ResidualBasedBlockBuilderAndSolver<TSparseSpace, TDenseSpace, TLinearSolver>;
+
+    /// Definition of the classes from the base class
+    using TSchemeType = typename BaseType::TSchemeType;
+    using TSystemMatrixType = typename BaseType::TSystemMatrixType;
+    using TSystemVectorType = typename BaseType::TSystemVectorType;
+    using LocalSystemVectorType = typename BaseType::LocalSystemVectorType;
+    using LocalSystemMatrixType = typename BaseType::LocalSystemMatrixType;
+    using NodesArrayType = typename BaseType::NodesArrayType;
+    using ElementsArrayType = typename BaseType::ElementsArrayType;
+    using ConditionsArrayType = typename BaseType::ConditionsArrayType;
+
+    /// Additional definitions
+    using ElementsContainerType = PointerVectorSet < Element, IndexedObject>;
+    using EquationIdVectorType = Element::EquationIdVectorType;
+
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /**
+     * @brief Default constructor
+     */
+    explicit ResidualBasedBlockBuilderAndSolverWithMassAndDamping() : BaseType()
+    {
+    }
+
+    /**
+     * @brief Default constructor. (with parameters)
+     */
+    explicit ResidualBasedBlockBuilderAndSolverWithMassAndDamping(
+        typename TLinearSolver::Pointer pNewLinearSystemSolver,
+        Parameters ThisParameters
+        ) : BaseType(pNewLinearSystemSolver)
+    {
+        // Validate and assign defaults
+        ThisParameters = this->ValidateAndAssignParameters(ThisParameters, this->GetDefaultParameters());
+        this->AssignSettings(ThisParameters);
+    }
+
+    /**
+     * @brief Default constructor.
+     */
+    explicit ResidualBasedBlockBuilderAndSolverWithMassAndDamping(typename TLinearSolver::Pointer pNewLinearSystemSolver)
+        : BaseType(pNewLinearSystemSolver)
+    {
+    }
+
+    /** Destructor.
+     */
+    ~ResidualBasedBlockBuilderAndSolverWithMassAndDamping() override
+    {
+    }
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * @brief Function to perform the build of the RHS. The vector could be sized as the total number
+     * of dofs or as the number of unrestrained ones
+     * @param pScheme The integration scheme considered
+     * @param rModelPart The model part of the problem to solve
+     * @param rA The LHS matrix
+     * @param rb The RHS vector
+     */
+    void Build(
+        typename TSchemeType::Pointer pScheme,
+        ModelPart& rModelPart,
+        TSystemMatrixType& rA,
+        TSystemVectorType& rb) override
+    {
+        KRATOS_TRY
+
+        KRATOS_ERROR_IF(!pScheme) << "No scheme provided!" << std::endl;
+
+        // Getting the elements from the model
+        const int nelements = static_cast<int>(rModelPart.Elements().size());
```

Comment:
Avoid repeating the type (the `static_cast` is already explicit about it):
```suggestion
        const auto nelements = static_cast<int>(rModelPart.Elements().size());
```